generic/contracts/StringList.sol

```diff
@@ -0,0 +1,112 @@
+//SPDX-License-Identifier: Unlicense
+pragma solidity ^0.8.0;
+
+import "@openzeppelin/contracts/access/Ownable.sol";
+
+contract StringList is Ownable {
+    /// Events
+    event Add(address indexed _caller, string _value);
+    event Remove(address indexed _caller,  string _value);
+
+    string constant PLACE_HOLDER = "____INVALID_PLACE_HOLER";
+
+    /// Errors
+    string constant ERROR_VALUE_NOT_PART_OF_THE_LIST = "ERROR_VALUE_NOT_PART_OF_THE_LIST";
+    string constant ERROR_VALUE_PART_OF_THE_LIST = "ERROR_VALUE_PART_OF_THE_LIST";
+    string constant ERROR_INVALID_INDEX = "ERROR_INVALID_INDEX";
+    string constant ERROR_INVALID_VALUE = "ERROR_INVALID_VALUE";
+
+    /// State
+    string public name;
+    string[] public values;
+    mapping(string => uint256) internal indexByValue;
+
+    /**
+     * @dev Initialize contract
+     * @notice Create a new list with name `_name`
+     * @param _name The list's display name
+     */
+    constructor(string memory _name) {
+        name = _name;
+
+        // Invalidate first position
+        values.push(PLACE_HOLDER);
+    }
+
+    /**
+     * @dev Add a value to the  list
+     * @notice Add "`_value`" to the string` list.
+     * @param _value String value to remove
+     */
+    function add(string calldata _value) external onlyOwner {
+        // Check if the value is part of the list
+        require(indexByValue[_value] == 0, ERROR_VALUE_PART_OF_THE_LIST);
+
+        // Check if the value is not the placeholder
+        require(keccak256(abi.encodePacked(_value)) != keccak256(abi.encodePacked(PLACE_HOLDER)), ERROR_INVALID_VALUE);
+
+        _add(_value);
+    }
+
+    /**
+     * @dev Remove a value from the list
+     * @notice Remove "`_value`" from the `self.symbol(): string` list
```

Comment:
`self.symbol(): string` would this refer to the values array?

Reply:
My bad. The aragon list also had a "Symbol" parameter in the constructor parameter. It was a string but not used anywhere. I removed it because it was not necessary.